Lib/test/test_pty.py

```diff
@@ -176,53 +176,54 @@ def test_openpty(self):
             # " An optional feature could not be imported " ... ?
             raise unittest.SkipTest("Pseudo-terminals (seemingly) not functional.")
 
-        self.assertTrue(os.isatty(slave_fd), "slave_fd is not a tty")
-
-        if mode:
-            self.assertEqual(tty.tcgetattr(slave_fd), mode,
-                             "openpty() failed to set slave termios")
-        if new_stdin_winsz:
-            self.assertEqual(_get_term_winsz(slave_fd), new_stdin_winsz,
-                             "openpty() failed to set slave window size")
-
-        # Solaris requires reading the fd before anything is returned.
-        # My guess is that since we open and close the slave fd
-        # in master_open(), we need to read the EOF.
-        #
-        # NOTE: the above comment is from an older version of the test;
-        # master_open() is not being used anymore.
-
-        # Ensure the fd is non-blocking in case there's nothing to read.
-        blocking = os.get_blocking(master_fd)
         try:
```

Comment:
Instead of the big indent, you could use `self.addCleanup(os.close, master_fd)` and `self.addCleanup(os.close, slave_fd)`

Reply:
TIL! Thank you for the suggestion!